### PR TITLE
Adds Mittelmann benchmark reference

### PR DESCRIPTION
**Pull request by merschformann**
# Description

I didn't make it in time to incorporate @jajhall's comment (see [here](https://github.com/ebrahimpichka/awesome-optimization/pull/1#issuecomment-1791685296)) from the other PR (#1). :sweat_smile: 

I think it makes sense to reference these key benchmarks for people interested in a well-known comparison of (some of) the listed solvers.

I introduced a section for the solvers alone to make the benchmark part feel less out-of-place. I hope this makes sense to you.

And like always, feel free to change as you see fit or reject. :blush:

### CODE REVIEW

**Comment by jajhall** [comment]
Thanks. I know that the folk behind SCIP are (also) keen that the poor performance of GLPK/lpsolve is publicised, because they are very popular but, performance-wise, give the wrong idea of what open-source is capable of solving.

**Comment by ebrahimpichka** [comment]
Thanks @merschformann, actually I added a whole separate topic header for solvers just before checking this PR, just as you aslo did in this PR. Could you re-sync your fork then add the benchmarks?

**Comment by merschformann** [comment]
I adapted to the new structure with quite some nesting. I still like how it's represented when rendered, but feel free to change it around again. :blush:

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ This list tries to cover topics on discrete and combinatorial optimization, oper
     1. [Combinatorial Optimization](#combinatorial-optimization)
     1. [Stochastic Optimization](#stochastic-optimization)
 3. [**Solvers, Tools, and Libraries**](#solvers-tools-and-libraries)
-    1. [Licensed/Commercial Solvers](#licensed-commercial-solvers)
+    1. [Licensed/Commercial Solvers](#licensedcommercial-solvers)
```
Comment (merschformann) [inline README.md:22]: Fixed a link here. Apparently, we can just omit the slash and it works. :shrug: 
Found this [here](https://stackoverflow.com/a/43894240/6795416).

Comment (ebrahimpichka) [reply README.md:22]: Oh didn't notice that. thanks.

**Pull request merged**